packages/core/src/components/hotkeys/hotkeysEvents.ts

```diff
@@ -53,32 +53,50 @@ export class HotkeysEvents {
     }
 
     public handleKeyDown = (e: KeyboardEvent) => {
-        if (this.isTextInput(e) || isHotkeysDialogShowing()) {
+        const isTextInput = this.isTextInput(e);
+
+        if (isHotkeysDialogShowing()) {
             return;
         }
 
         const combo = getKeyCombo(e);
 
-        if (comboMatches(parseKeyCombo(SHOW_DIALOG_KEY), combo)) {
+        if (!isTextInput && comboMatches(parseKeyCombo(SHOW_DIALOG_KEY), combo)) {
             showHotkeysDialog(this.actions.map((action) => action.props));
             return;
         }
 
         for (const action of this.actions) {
-            if (comboMatches(action.combo, combo)) {
+            const shouldIgnore = (isTextInput && !action.props.allowInInput) || action.props.disabled;
+            if (comboMatches(action.combo, combo) && !shouldIgnore) {
```

Comment:
put `!shouldIgnore` check first to avoid more expensive check if we already know to ignore it.